### ISSUE TITLE
Cannot find rxjs-imports.ts in the compiled program

### ISSUE THREAD

**Issue by kreuzerk** [cartant/rxjs-tslint-rules#15]
Hello

I have setup rxjs-tslint-rules with Angular CLI 1.4.7. I have followed the README with the Angular Gotchas as well as this blogpost https://christianliebel.com/2017/07/import-rxjs-correctly/.

I ended up with the following file updates:

- In tsconfig.json I added the following:
```json
"rules": {
    "rxjs-add": {
      "options": [
        {
          "allowElsewhere": false,
          "allowUnused": false,
          "file": "./src/rxjs-imports.ts"
        }
      ],
      "severity": "error"
    }
  }
```

In my tslint.json I added:
```json
"extends": [
    "rxjs-tslint-rules"
  ],
  "rules": {
    "rxjs-add": {
      "severity": "error",
      "options": [
        {
          "allowElsewhere": false,
          "allowUnused": false,
          "file": "./src/rxjs-imports.ts"
        }
      ]
    },
    "rxjs-no-unused-add": {
      "severity": "error"
    },
}
```

Then I created a rxjs-imports.ts file:
```javascript
import 'rxjs/add/observable/interval';
import 'rxjs/add/operator/map';
```

I then imported this file in my main.ts with the following import statement:
```javascript 
import 'rxjs-imports.ts'
```
Furthermore I had to pass --type-check to the linting task. When I run npm run lint --type-check I get the following error:

```
> ng lint --type-check

Error: Cannot find '/users/kevinkreuzer/desktop/central/rxjscentral/src/rxjs-imports.ts' in the compiled program. Has it been imported?
    at Walker.findSourceFile (/Users/kevinkreuzer/Desktop/central/rxjsCentral/node_modules/rxjs-tslint-rules/dist/rules/rxjsAddRule.js:136:15)
    at Walker.walkFile (/Users/kevinkreuzer/Desktop/central/rxjsCentral/node_modules/rxjs-tslint-rules/dist/rules/rxjsAddRule.js:146:35)
    at Walker.onSourceFileEnd (/Users/kevinkreuzer/Desktop/central/rxjsCentral/node_modules/rxjs-tslint-rules/dist/rules/rxjsAddRule.js:48:31)
    at Walker.AddedWalker.visitNode (/Users/kevinkreuzer/Desktop/central/rxjsCentral/node_modules/rxjs-tslint-rules/dist/support/added-walker.js:41:18)
    at Walker.SyntaxWalker.walk (/Users/kevinkreuzer/Desktop/central/rxjsCentral/node_modules/tslint/lib/language/walker/syntaxWalker.js:24:14)
    at Rule.AbstractRule.applyWithWalker (/Users/kevinkreuzer/Desktop/central/rxjsCentral/node_modules/tslint/lib/language/rule/abstractRule.js:31:16)
    at Rule.applyWithProgram (/Users/kevinkreuzer/Desktop/central/rxjsCentral/node_modules/rxjs-tslint-rules/dist/rules/rxjsAddRule.js:15:21)
    at Linter.applyRule (/Users/kevinkreuzer/Desktop/central/rxjsCentral/node_modules/tslint/lib/linter.js:176:29)
    at /Users/kevinkreuzer/Desktop/central/rxjsCentral/node_modules/tslint/lib/linter.js:121:85
    at Object.flatMap (/Users/kevinkreuzer/Desktop/central/rxjsCentral/node_modules/tslint/lib/utils.js:151:29)

ERROR: /Users/kevinkreuzer/Desktop/central/rxjsCentral/src/rxjs-imports.ts[1, 1]: RxJS add import is unused: interval
ERROR: /Users/kevinkreuzer/Desktop/central/rxjsCentral/src/rxjs-imports.ts[2, 1]: RxJS add import is unused: map
ERROR: /Users/kevinkreuzer/Desktop/central/rxjsCentral/src/rxjs-imports.ts[2, 32]: file should end with a newline
```

The path where it is looking for the file seems to be correct. Am I missing something? I thought by including the rxjs-imports.ts in my main.ts this file should be included in the compiled program.

**Comment by cartant** [comment]
You'll need to use a relative path for this import:

```ts
import './rxjs-imports.ts'
```

**Comment by kreuzerk** [comment]
@cartant Thanks for the quick response. Yes you are absolutely right. But unfortunately that does not fix the issue.

**Comment by kreuzerk** [comment]
I have created a sample of this issue with the new Angular CLI. I have pushed the code to https://github.com/kreuzerk/centralRxJS. If you want you can have a quick look at it.

**Comment by cartant** [comment]
There are paths in there with differing cases:

    /users/kevinkreuzer/desktop/central/rxjscentral/src/rxjs-imports.ts

and:

    /Users/kevinkreuzer/Desktop/central/rxjsCentral/src/rxjs-imports.ts

What OS are you using?

**Comment by kreuzerk** [comment]
I am using a MacBook with Sierra

**Comment by cartant** [comment]
It looks like you need to include `rxjs-imports.ts` in the tests, too. In the [`README.md`](https://github.com/cartant/rxjs-tslint-rules#angularcli), I suggest adding it to `src/tsconfig.spec.json`, but it does not appear to be in [your file](https://github.com/kreuzerk/centralRxJS/blob/master/src/tsconfig.spec.json).

**Comment by cartant** [comment]
Closing this, as I'm assuming it's resolved. If it isn't, please let me know.

**Comment by kreuzerk** [comment]
Sorry for answering late. I updated the tsconfig.spec.json but I still get the same error. I would like to reopen this issue as it is still not working. 

I can not see what I am missing. Do you maybe have a working example with Angular CLI?

**Comment by cartant** [comment]
The repo now includes a working `@angular/cli` example.

A vanilla CLI app was added in [this commit](https://github.com/cartant/rxjs-tslint-rules/commit/72098955273c076e6cb5f2d22a8009194d5ac918). And its linting was configured in [this commit](https://github.com/cartant/rxjs-tslint-rules/commit/73872cc381765d2fcc9fc22cb686128b76dfd6fb).

**Comment by kreuzerk** [comment]
Great. Thanks a lot. I tried your example and it is working!
So there was something I did wrong. I will try to figure out what caused the problem.

Thank you!

**Comment by cartant** [comment]
No worries. Good luck.
